tests-utils/src/main/scala/tech/beshu/ror/utils/containers/generic/EsImage.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.utils.containers.generic
+
+import com.typesafe.scalalogging.StrictLogging
+import org.testcontainers.images.builder.ImageFromDockerfile
+import org.testcontainers.images.builder.dockerfile.DockerfileBuilder
+import tech.beshu.ror.utils.containers.DockerfileBuilderOps._
+import tech.beshu.ror.utils.containers.generic.EsContainer.Config
+import tech.beshu.ror.utils.misc.Version
+
+trait EsImage[CONFIG <: EsContainer.Config] extends StrictLogging {
+
+  protected def copyNecessaryFiles(builder: DockerfileBuilder, config: CONFIG): DockerfileBuilder = builder
+
+  protected def entry(config: CONFIG): ImageFromDockerfile = new ImageFromDockerfile()
+
+  protected def install(builder: DockerfileBuilder, config: CONFIG): DockerfileBuilder = builder
+
+  def create(config: CONFIG): ImageFromDockerfile = {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method is longer than 50 lines](https://app.codacy.com/manual/scarduzio/elasticsearch-readonlyrest-plugin/pullRequest?prid=5012330)